package/scripts/ambari_helpers.py

```diff
@@ -43,7 +43,20 @@ def cdap_config(name=None):
        content=InlineTemplate(params.cdap_env_sh_template)
   )
 
-  cleanup_opts()
+  if name == 'auth':
+    dirname = 'security'
+  elif name == 'router':
+    dirname = 'gateway'
+  else:
+    dirname = name
+
+  cleanup_opts(dirname)
+
+  # Copy logback.xml and logback-container.xml
+  for i in 'logback.xml', 'logback-container.xml':
+    no_op_test = 'ls ' + params.cdap_conf_dir + '/' + i + ' 2>/dev/null'
+    Execute('cp -f /etc/cdap/conf.dist/' + i + ' ' + params.cdap_conf_dir, not_if=no_op_test)
```

Comment:
So reading from left to right to back to the left to fill in the variable that you read from the right is easier than reading from left to right, sequentially? Could you give me an example?